### PR TITLE
Version comparison fixes

### PR DESCRIPTION
**Pull request by nd-novorender**
- Auto enable/open property tree for version comparison scene
- Expand Compare node in property tree
- Home button transitions into original state, so proper camera reset and same widget things

https://trello.com/c/xBhdpag2/962-more-version-comparison-fixes

### CODE REVIEW

File: src/features/home/useResetView.ts [hunk 1]
```diff
@@ -68,6 +78,15 @@ export function useResetView() {
                     ),
                 ),
             );
+
+            if (
+                db &&
+                view?.renderState.scene?.url &&
+                isVersionComparisonUrl(view.renderState.scene.url) &&
+                resetVersionComparisonPropertyTree
```
Comment (barabadzhi) [inline src/features/home/useResetView.ts:86]: Short circuit by putting `resetVersionComparisonPropertyTree` check first.
No need to do anything else if the flag is `false`.

File: src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+async function selectVersionComparisonProperties(db: ObjectDB, dispatch: ReturnType<typeof useAppDispatch>) {
+    const properties = ["Changed", "Deleted", "New"];
+    const groups = await Promise.all(
+        properties
+            .map(async (propValue, index) => {
+                const resp = db.search(
+                    { searchPattern: [{ property: "Compare", value: propValue, exact: true }] },
+                    undefined,
```
Comment (barabadzhi) [inline src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts:35]: Is explicit `undefined` strictly needed here?

File: src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                );
+                const ids: number[] = [];
+                for await (const obj of resp) {
+                    ids.push(obj.id);
+                }
```
Comment (barabadzhi) [inline src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts:40]: Is this a
```js
const ids = Promise.all(resp).then(objs => objs.map(obj => obj.id);
```
?

File: src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                if (ids.length === 0) {
+                    return;
+                }
+                return {
+                    group: {
+                        propertyValue: propValue,
+                        ids: Object.fromEntries(ids.map((id) => [id, id])),
```
Comment (barabadzhi) [inline src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts:47]: Or even
```js
const ids = (await Promise.all(resp)).reduce((ids, obj) => ({ ...ids, [obj.id]: obj.id }), {});
```
No need for intermediate array then.

File: src/features/render/hooks/useHandleInit.ts [hunk 5]
```diff
@@ -223,10 +224,10 @@ export function useHandleInit() {
                     }),
                 );
 
-                const isVersionComparison = parentSceneId.includes("/version-comparison/");
-                if (isVersionComparison && db && !sceneCamera) {
+                const isVersionComparison = isVersionComparisonUrl(parentSceneId);
+                if (db && isVersionComparison && !sceneCamera) {
```
Comment (barabadzhi) [inline src/features/render/hooks/useHandleInit.ts:228]: Order?
Can `!sceneCamera` shortcut earlier?
We can then do `isVersionComparisonUrl(parentSceneId)` inline (as the result is not used elsewhere), skipping the call if `sceneCamera || !db`.

File: src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts (hunk 2 above)
Comment (nd-novorender) [reply src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts:35]: Yeap, we don't need abort controller here and `search` signature requires this argument

File: src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts (hunk 3 above)
Comment (nd-novorender) [reply src/features/propertyTree/hooks/useSetupVersionComparisonPropertyTree.ts:40]: `resp` is `AsyncIterableIterator`, I don't think `Promise.all` works with them this way

**Pull request merged**